src/rules/font_face.rs

```diff
@@ -104,20 +110,46 @@ pub struct UrlSource<'i> {
   pub url: Url<'i>,
   /// Optional `format()` function.
   #[cfg_attr(feature = "serde", serde(borrow))]
-  pub format: Option<Format<'i>>,
+  pub format: Option<FontFormat<'i>>,
+  /// Optional `tech()` function.
+  pub tech: Vec<FontTechnology>,
 }
 
 impl<'i> Parse<'i> for UrlSource<'i> {
   fn parse<'t>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i, ParserError<'i>>> {
     let url = Url::parse(input)?;
 
+    let mut format_location: Option<SourceLocation> = None;
     let format = if input.try_parse(|input| input.expect_function_matching("format")).is_ok() {
-      Some(input.parse_nested_block(Format::parse)?)
+      format_location = Some(input.current_source_location());
+      Some(input.parse_nested_block(FontFormat::parse)?)
     } else {
       None
     };
 
-    Ok(UrlSource { url, format })
+    let tech = if input.try_parse(|input| input.expect_function_matching("tech")).is_ok() {
+      if let Some(location) = format_location {
+        // parser error
+        let tech_location = input.current_source_location();
+        if tech_location.line < location.line {
+          return Err(ParseError {
+            kind: ParseErrorKind::Basic(BasicParseErrorKind::AtRuleBodyInvalid),
+            location: input.current_source_location(),
+          });
+        } else if tech_location.line == location.line && tech_location.column < location.column {
+          return Err(ParseError {
+            kind: ParseErrorKind::Basic(BasicParseErrorKind::AtRuleBodyInvalid),
+            location: input.current_source_location(),
+          });
+        }
+      }
```

Comment:
How could these checks ever happen? `format_location` will always be less than `tech_location` since you already parsed it before if it exists.